### PR TITLE
Added market approval server backup script

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Added market approval server backup script

### CODE REVIEW

File: market_approval_server/backup_config.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+
+def _fetch_data(endpoint_url: str) -> Optional[Dict[str, Any]]:
+    response = requests.get(endpoint_url)
+    if response.status_code == 200:
+        return response.json()
+    print(
+        f"Failed to fetch data from {endpoint_url}. Status code: {response.status_code}"
+    )
+    return None
```
Comment (Adamantios) [inline market_approval_server/backup_config.py:37]: Add some retry logic here.

**Review by Adamantios — APPROVED** [review]

File: market_approval_server/backup_config.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def _process_endpoints(url: str) -> Optional[Dict[str, Any]]:
+    approved_markets = _fetch_data(f"{url}/approved_markets")
+    rejected_markets = _fetch_data(f"{url}/rejected_markets")
+    proposed_markets = _fetch_data(f"{url}/proposed_markets")
+    processed_markets = _fetch_data(f"{url}/processed_markets")
+
+    if approved_markets and rejected_markets and proposed_markets and processed_markets:
```
Comment (Adamantios) [inline market_approval_server/backup_config.py:53]: So if `approved_markets` is `None` then we fetched everything for no reason. Would be better to terminate immediately.

File: market_approval_server/backup_config.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+            print(f"Retrying in {RETRY_DELAY_SECONDS} seconds...")
+            time.sleep(RETRY_DELAY_SECONDS)
+    return None
+
+
+def _process_endpoints(url: str) -> Optional[Dict[str, Any]]:
+    approved_markets = _fetch_data(f"{url}/approved_markets")
+    rejected_markets = _fetch_data(f"{url}/rejected_markets")
+    proposed_markets = _fetch_data(f"{url}/proposed_markets")
+    processed_markets = _fetch_data(f"{url}/processed_markets")
```
Comment (Adamantios) [inline market_approval_server/backup_config.py:51]: Could try something like this to return earlier:

```suggestion
    _backup_data = {"api_keys": {                "454d31ff03590ff36836e991d3287b23146a7a84c79d082732b56268fe472823": "default_user"}}
    for market_name in market_names:
        res = _fetch_data(f"{url}/market_name")
        if res is None:
            return None
        _backup_data[market_name] = res.get(market_name, {})
```

File: market_approval_server/backup_config.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                "454d31ff03590ff36836e991d3287b23146a7a84c79d082732b56268fe472823": "default_user"
+            },
+            "approved_markets": approved_markets.get("approved_markets", {}),
+            "processed_markets": processed_markets.get("processed_markets", {}),
+            "proposed_markets": proposed_markets.get("proposed_markets", {}),
+            "rejected_markets": rejected_markets.get("rejected_markets", {}),
+        }
+        return _backup_data
+
+    return None
```
Comment (Adamantios) [inline market_approval_server/backup_config.py:65]: Could try something like this to return earlier:

```suggestion
    _backup_data = {"api_keys": {                "454d31ff03590ff36836e991d3287b23146a7a84c79d082732b56268fe472823": "default_user"}}
    for market_name in market_names:
        res = _fetch_data(f"{url}/market_name")
        if res is None:
            return None
        _backup_data[market_name] = res.get(market_name, {})
    return _backup_data
```

File: market_approval_server/backup_config.py (hunk 2 above)
Comment (jmoreira-valory) [reply market_approval_server/backup_config.py]: fixed

File: market_approval_server/backup_config.py (hunk 4 above)
Comment (jmoreira-valory) [reply market_approval_server/backup_config.py]: fixed

**Pull request merged**